src/main/groovy/edu/oregonstate/mist/trailsapi/db/TrailDAO.groovy

```diff
@@ -51,4 +63,52 @@ interface TrailDAO extends Closeable {
 				WHERE DIFFICULTY_COLOR = :trail.difficulty)
 	""")
 	Boolean getConflictingTrails(@BindBean("trail") Trail trail)
+
+	/**************************************************************************************************
+	GET /trails
+	**************************************************************************************************/
+	@SqlQuery("""
+		SELECT
+			TRAILS.ID,
+			TRAILS.NAME,
+			TRAIL_DIFFICULTIES.DIFFICULTY_COLOR,
+			TRAILS.POLYLINE,
+			TRAILS.ZIPCODE,
+			TRAILS.SMALLDROP,
+			TRAILS.LARGEDROP,
+			TRAILS.WOODRIDE,
+			TRAILS.SKINNY,
+			TRAILS.LARGEJUMP,
+			TRAILS.SMALLJUMP,
+			TRAILS.GAP
+		 FROM TRAILS
+		 LEFT JOIN TRAIL_DIFFICULTIES ON TRAILS.DIFFICULTY_ID = TRAIL_DIFFICULTIES.DIFFICULTY_ID
+		 WHERE UPPER(REPLACE(REPLACE(TRAILS.NAME, ' ', ''), '''', ''))
+			= UPPER(REPLACE(REPLACE(:name, ' ', ''), '''', '')) OR :name IS NULL
+		 	AND (UPPER(TRAIL_DIFFICULTIES.DIFFICULTY_COLOR) = UPPER(:difficulty) OR :difficulty IS NULL)
+			AND (TRAILS.DIFFICULTY_ID <= (SELECT DIFFICULTY_ID FROM TRAIL_DIFFICULTIES WHERE
```

Comment:
It looks like you're using the trail_difficulties.difficulty_id as a ranking field too. No need to change it, but in the future for something like this it might be better to have a separate column in the database for "rank". This would allow you to add new difficulties somewhere in the middle (maybe a blue-green rating?) without needing to change the IDs of the other difficulty levels. You would just need to change the values of the ranking columns to reassign them for each difficulty.

Reply:
That would be cool, it would really take advantage of having trail difficulties in a separate table. I'll keep that in mind next time I'm doing database design.